modules/navier_stokes/include/userobjects/NSFVRhieChowInterpolator.h

```diff
@@ -0,0 +1,158 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "UserObject.h"
+#include "TheWarehouse.h"
+#include "SubProblem.h"
+#include "MooseApp.h"
+#include "INSFVAttributes.h"
+#include "FVElementalKernel.h"
+#include "MooseUtils.h"
+#include "FVUtils.h"
+
+// Do we really need all this
+#include "NeighborCoupleable.h"
+#include "TwoMaterialPropertyInterface.h"
+#include "NeighborMooseVariableInterface.h"
+#include "NeighborCoupleableMooseVariableDependencyIntermediateInterface.h"
+
+#include <vector>
+#include <set>
+
+class INSFVVelocityVariable;
+class INSFVPressureVariable;
+
+/**
+ * An object that can compute interpolated velocities on faces
+ */
+class NSFVRhieChowInterpolator : public UserObject,
+                                 public TwoMaterialPropertyInterface,
+                                 public NeighborMooseVariableInterface<Real>,
+                                 public NeighborCoupleableMooseVariableDependencyIntermediateInterface,
+                                 public BlockRestrictable
```

Comment:
This inheritance chain looks remarkably similar to `InternalSideUserObject`. Can we just use that?

Reply:
Sounds good. InternalSide doesn't feel like it refers to an element face but in the end I think it does!